### PR TITLE
Add milmove-app-browsers Dockerfile

### PR DESCRIPTION
**Pull request by gidjin**
Add Dockerfile for creating a milmove-app-browsers image with chrome installed. This is to be used with the Loki tests for Storybook stories.

I used the example from the CircleCI Dockerfile for setting this up.
https://github.com/CircleCI-Public/circleci-dockerfiles/blob/master/node/images/10.19.0-buster/browsers/Dockerfile

### CODE REVIEW

**Comment by gidjin** [comment]
@chrisgilmerproj I modified the `test` file, I also updated the list of "special" branches and the `build` script so that it would be built along with the rest. Let me know if I missed anything else for adding this new image.

File: milmove-app-browsers/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+#ENV CHROME_PATH=/opt/google/chrome
+
+# ensure that the build agent doesn't override the entrypoint
+LABEL com.circleci.preserve-entrypoint=true
+
+USER circleci
+
+ENTRYPOINT ["/docker-entrypoint.sh"]
+CMD ["/bin/sh"]
```
Comment (chrisgilmerproj) [inline milmove-app-browsers/Dockerfile]: What does this mean? That you're passing `/bin/sh` through the `/docker-entrypoint.sh`? That's interesting to me. I don't know what that other script is doing.

File: milmove-app-browsers/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    && rm -rf chromedriver_linux64.zip \
+    && sudo mv chromedriver /usr/local/bin/chromedriver \
+    && sudo chmod +x /usr/local/bin/chromedriver \
+    && chromedriver --version
+
+# start xvfb automatically to avoid needing to express in circle.yml
+ENV DISPLAY :99
+RUN printf '#!/bin/sh\nXvfb :99 -screen 0 1280x1024x24 &\nexec "$@"\n' > /tmp/entrypoint \
+  && chmod +x /tmp/entrypoint \
+        && sudo mv /tmp/entrypoint /docker-entrypoint.sh
```
Comment (chrisgilmerproj) [inline milmove-app-browsers/Dockerfile]: Looks like there may be a tab/spaces issue here. Can you update it?

File: test [hunk 3]
```diff
@@ -34,6 +34,17 @@ function test_milmove_app() {
   echo "Passed ${tag}"
 }
 
+function test_milmove_app_browsers() {
+  tag=milmove-app-browsers
+  echo "Testing ${tag} Dockerfile"
+
+  docker run -it "milmove/circleci-docker:${tag}" node --version
+  docker run -it "milmove/circleci-docker:${tag}" yarn --version
```
Comment (chrisgilmerproj) [inline test]: You don't have to test node and yarn if we're building off a previous image which has those where they are already being tested.

File: test [hunk 4]
```diff
@@ -34,6 +34,17 @@ function test_milmove_app() {
...
+  docker run -it "milmove/circleci-docker:${tag}" google-chrome --version
```
Comment (chrisgilmerproj) [inline test]: Is there a way to check the validity of the chromedriver install as well?

File: milmove-app-browsers/Dockerfile (hunk 1 above)
Comment (gidjin) [reply milmove-app-browsers/Dockerfile]: I copied those bits from the CircleCI ones. As I understand `docker-entrypoint.sh` is created during the creation of the image, see line 58 of this file. What it does is start the `Xvfb` service in the background and then calls `exec "$@"` to run whatever is passed to it. The `CMD ["/bin/sh"]` is the default shell for docker as I recall, I'm happy to adjust the CMD if needed but the entrypoint needs to start the `Xvfb` service for Chrome to work. I have seen the exec trick to do at runtime setup in a container in the past. So it wasn't that unusual to me.

**Pull request merged**